src/tables.jl

```diff
@@ -98,8 +102,19 @@ struct DimTable <: AbstractDimTable
     dimarraycolumns::Vector{AbstractVector}
 end
 
-function DimTable(s::AbstractDimStack; mergedims=nothing)
+function DimTable(s::AbstractDimStack; 
+    mergedims=nothing,
+    preservedims=nothing,
+)
     s = isnothing(mergedims) ? s : DD.mergedims(s, mergedims)
+    s = if isnothing(preservedims) 
+        s
+    else
+        maplayers(s) do A
+            S = DimSlices(A; dims=preservedims)
```

Comment:
Maybe this should be 
```suggestion
            S = DimSlices(A; dims=otherdims(A, preservedims))
```